backend/internal/handlers/login.go

```diff
@@ -44,7 +42,38 @@ func (lh *LoginHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// TODO: remove
-	fakeAuthLogic(&body, w, r)
-	return
+	account, err := lh.db.GetAccount(body.Email)
+
+	// Handle errors from attempting to retrieve the account from the database
+	if err != nil {
+		log.Println(err)
+		if err == sql.ErrNoRows {
+			// No record with the given email address exists
+			http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+			return
+		}
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	// Account retrieved, check password
+	if !auth.CheckPasswordHash(body.Password, account.PasswordHash) {
+		// Invalid password, unauthorized
+		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
+		return
+	}
+
+	// Valid password, create new session
+	session, err := lh.sm.CreateSession(account)
+	if err != nil {
+		log.Println(err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(loginResponseBody{session.SessionID, session.Account.Plan}); err != nil {
+		log.Println(err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
```

Comment:
nit: when you write to http.ResponseWriter for the first time, it automatically writes the status code first (200 OK, unless WriteHeader was called before).
so you can't write a different status after you've started writing the response body.

it's ok to just log the error but not send anything new in response.